### DISCUSSION TITLE
Database error

### DISCUSSION THREAD

**Discussion by alezanott** [teslamate-org/teslamate#3325] [Q&A]
When accessing the dashboards on Grafana I get this error `db query error: pq: the database system is in recovery mode`. What can I do to fix it? I think that this regards the Postegre SQL database.

**Comment by cwanja** [comment]
That is referring to the PostgreSQL database. 

What do the logs say? Have you tried restarting the database container or the entire TeslaMate service?

**Reply by alezanott** [reply]
I've run now `docker system prune --all` and for some second I was able to see the charges dashboard on Grafana. Now I've got this new error `db query error: pq: syntax error at or near "AND"`.
What I'm trying to do is making a backup as I want to transfer it to a new RPI. The first time I've created the backup file, when restoring it I got a DB error regarding the charges. As this the new install is not running.

EDIT: now the Teslamate webpage is not working but on Grafana I can see the charge dashboard... I don't understand what's happening. If I click on the price of a charge it brings me to the Teslamate webpage, at this point nothing is working.

**Reply by alezanott** [reply]
I'm running now Teslamate in verbose mode, so basically running docker-compose without the -d. I see clearly that it crashes and it restarts all the process.

**Reply by cwanja** [reply]
Have you tried dropping the database volumes and re-establishing it? https://docs.teslamate.org/docs/maintenance/upgrading_postgres/

**Reply by alezanott** [reply]
Actually I can't drop the database because I can't make a backup. Everytime I try to make a backup the process get stuck. So basically I'm inside a loophole that I can't exit.

**Reply by cwanja** [reply]
> What I'm trying to do is making a backup as I want to transfer it to a new RPI. The first time I've created the backup file, when restoring it I got a DB error regarding the charges. As this the new install is not running.

You said you have a backup? And if this instance has not got up and running, there is no missing data in the database than what you have in your backup. 

If that is all true, drop the entire installation of TeslaMate and start over.

**Reply by alezanott** [reply]
Actually when importing this backup on the new RPI I get an error, so I think that is corrupted. I don't have a complete backup atm.
It's data since 2021, for this reason I wanted to transfer all the old data to the new RPI... But I understand that if so corrupted is better to make a new installation.

**Reply by cwanja** [reply]
It sounds like your new installation is actually borked. 

Do you have a backup? It seems like it. And if so (or even if not), do you have the old instance of TeslaMate still running?

If one of those is yes, completely remove the new instance of TeslaMate and try to start over.

**Reply by alezanott** [reply]
Basically the old instance of Teslamate is still running on my RPI 3. On my new RPI 4 I have just made a fresh install, when importing the backup basically it gets stuck with an error.

This is the error that I get now when backupping: 
```
pg_dump: WARNING:  terminating connection because of crash of another server process
DETAIL:  The postmaster has commanded this server process to roll back the current transaction and exit, because another server process exited abnormally and possibly corrupted shared memory.
HINT:  In a moment you should be able to reconnect to the database and repeat your command.
pg_dump: error: Dumping the contents of table "positions" failed: PQgetCopyData() failed.
pg_dump: error: Error message from server: server closed the connection unexpectedly
	This probably means the server terminated abnormally
	before or while processing the request.
pg_dump: error: The command was: COPY public.positions (id, date, latitude, longitude, speed, power, odometer, ideal_battery_range_km, battery_level, outside_temp, elevation, fan_status, driver_temp_setting, passenger_temp_setting, is_climate_on, is_rear_defroster_on, is_front_defroster_on, car_id, drive_id, inside_temp, battery_heater, battery_heater_on, battery_heater_no_power, est_battery_range_km, rated_battery_range_km, usable_battery_level, tpms_pressure_fl, tpms_pressure_fr, tpms_pressure_rl, tpms_pressure_rr) TO stdout;
```

Still I get a backup file of 425 Mb.
Can you please help me @cwanja. Thank you!

**Comment by alexeiw123** [comment]
I'm also seeing this error since running `docker compose pull` and `docker compose up -d`.

My teslamate container is crashing usually within a few seconds of starting.

I set up a new install which was OK, until I tried to restore my backup, which errors out same as above and from then the teslamate container starts crashing again.
